### DISCUSSION TITLE
2023/09/17/common-ls/

### DISCUSSION THREAD

**Discussion by giscus** [argvchs/giscus-comments#31] [Announcements]
# 2023/09/17/common-ls/

https://argvchs.netlify.app/2023/09/17/common-ls/

**Comment by Rickyxrc** [comment]
感觉 encode/decode 功能被利用之后会干坏事，还是建议只运行信任的脚本。

还有就是这种规范如果能写成解析器（tampermonkey那种？）的形式就更好了，可以严控权限。

**Reply by argvchs** [reply]
~~但是写这个东西不就是为了干坏事吗~~（

**Reply by Rickyxrc** [reply]
当别人看到你被层层加密的代码之后还敢运行吗（

**Comment by immccn123** [comment]
`new Function(code)` 可以解决 eval 的问题

https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Function

副作用是只能访问 global scope

**Reply by immccn123** [reply]
应当是，可以解决 eval 性能降低的问题

**Reply by immccn123** [reply]
似乎可以解决 安全策略 的问题。

**Reply by immccn123** [reply]
经查证，不行。

**Reply by immccn123** [reply]
又发现了一种人类智慧方法解决这个问题。

```js
/** @param code {string}  */
function DOMEval(code) {
    let el = document.createElement("script");
    el.innerText = code;
    document.body.appendChild(el);
}
```

**Reply by argvchs** [reply]
%%%
